pydantic/_internal/_std_types_schema.py

```diff
@@ -109,67 +116,82 @@ def decimal_schema(_schema_generator: GenerateSchema, _decimal_type: type[Decima
             core_schema_override=lambda: decimal_validator.json_schema_override_schema()
         ),
     )
-    return core_schema.function_after_schema(
-        core_schema.union_schema(
-            core_schema.is_instance_schema(Decimal, json_types={'int', 'float'}),
-            core_schema.int_schema(),
-            core_schema.float_schema(),
-            core_schema.str_schema(strip_whitespace=True),
-            strict=True,
+    return core_schema.lax_or_strict_schema(
+        lax_schema=core_schema.function_after_schema(
+            core_schema.union_schema(
+                core_schema.is_instance_schema(Decimal, json_types={'int', 'float'}),
+                core_schema.int_schema(),
+                core_schema.float_schema(),
+                core_schema.str_schema(strip_whitespace=True),
+                strict=True,
+            ),
+            decimal_validator,
+            metadata=metadata,
         ),
-        decimal_validator,
+        strict_schema=core_schema.is_instance_schema(Decimal),
```

Comment:
to support max/min/nan checks, I think this needs to be
```suggestion
        strict_schema=core_schema.function_after_schema(
            core_schema.is_instance_schema(Decimal, json_types={'int', 'float'}),
            decimal_validator,
            metadata=metadata,
        ),
```